crates/solvers/src/domain/solution.rs

```diff
@@ -22,7 +22,9 @@ pub struct Solution {
     pub id: Id,
     pub prices: ClearingPrices,
     pub trades: Vec<Trade>,
+    pub pre_interactions: Vec<InteractionData>,
     pub interactions: Vec<Interaction>,
+    pub post_interactions: Vec<InteractionData>,
```

Comment:
It would lead to this:

```suggestion
    pub post_interactions: Vec<eth::Interaction>,
```

Reply:
But `eth::Interaction` is part of the driver crate, and the `solver` crate should not depend on the driver, right? Do you want me to extract this type into a shared crate, like `model`?

Reply:
Oh, you are completely right (I missed that this was a different create). We could still add this type in the eth module of the solvers crate but not super important.